ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime/HttpExtensions.cs

```diff
@@ -164,5 +164,46 @@ public static JObject ToJson(this HttpResponseHeaders headers)
                 return jObject;
             }
         }
+
+        /// <summary>
+        /// Serializes HttpResponseHeaders and HttpContentHeaders as Json dictionary.
+        /// </summary>
+        /// <param name="message">HttpResponseMessage</param>
+        /// <returns>Json string</returns>
+        public static JObject GetHeadersAsJson(this HttpResponseMessage message)
+        {
+            if (message == null)
+            {
+                return new JObject();
+            }
+
+            var jObject = new JObject();
+            foreach (var httpResponseHeader in message.Headers)
```

Comment:
It's not really documented anywhere but based on the source code it should never be null - https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/HttpResponseMessage.cs#L99